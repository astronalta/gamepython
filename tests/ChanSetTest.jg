# Copyright (c) 2010 Matt Fichman
# jgtest exec on

main() Int {

    ch1 = Chan[String](0)
    ch2 = Chan[String](0)

    cs = ChanSet[String,String]()
    cs.chan('ch1', ch1)
    cs.chan('ch2', ch2)

    coro = Coroutine() func() {
        for item in cs {
            Io::println("#{item.first} => #{item.second}")
        }
    }
    coro()

    ch1.put('hello')
    ch2.put('world')
    ch1.put('hello')
    ch2.put('world')
    ch2.put('world')
    ch2.put('world')

    Os::exit(0)
    ret 0
}
