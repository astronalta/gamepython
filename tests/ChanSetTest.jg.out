ch1 => hello
ch2 => world
ch1 => hello
ch2 => world
ch2 => world
ch2 => world
